lib/compiler/aro/aro/Preprocessor.zig

```diff
@@ -430,12 +430,12 @@ fn preprocessExtra(pp: *Preprocessor, source: Source) MacroError!TokenWithExpans
                         if_level = sum;
 
                         if (try pp.expr(&tokenizer)) {
-                            if_kind.set(if_level, until_endif);
+                            std.mem.writePackedIntNative(u2, &if_kind, if_level * 2, until_endif);
```

Comment:
`if_level * 2` will overflow if it's greater than 127. I have a PR up for aro https://github.com/Vexu/arocc/pull/787 that you can grab once it's approved and passing CI (it's waiting for `@splat` for arrays to be available in pre-built zig).